presto-main/src/main/java/com/facebook/presto/execution/executor/TaskHandle.java

```diff
@@ -106,9 +130,12 @@ public synchronized List<PrioritizedSplitRunner> destroy()
         destroyed = true;
 
         ImmutableList.Builder<PrioritizedSplitRunner> builder = ImmutableList.builderWithExpectedSize(runningIntermediateSplits.size() + runningLeafSplits.size() + queuedLeafSplits.size());
-        builder.addAll(runningIntermediateSplits);
-        builder.addAll(runningLeafSplits);
-        builder.addAll(queuedLeafSplits);
+        //To avoid queued split marked as completed splits to pollute the retryable splits
+        if (isShuttingDown.get()) {
```

Comment:
AGPQN : This check is a bit confusing. Can we call this `destroy` say if we want to do speculative execution and cancel/destroy the work which is independent of the notion of a shutdown.